src/Compiler/Driver/CompilerOptions.fs

```diff
@@ -1983,13 +1983,13 @@ let displayVersion tcConfigB =
 let miscFlagsBoth tcConfigB =
     [
         CompilerOption("nologo", tagNone, OptionUnit(fun () -> tcConfigB.showBanner <- false), None, Some(FSComp.SR.optsNologo ()))
-        CompilerOption("version", tagNone, OptionUnit(fun () -> displayVersion tcConfigB), None, Some(FSComp.SR.optsVersion ()))
+        CompilerOption("version", tagNone, OptionConsoleOnly(fun _ -> displayVersion tcConfigB), None, Some(FSComp.SR.optsVersion ()))
```

Comment:
@abelbraaksma @vzarytovskii this is the actual fix, everything else is refactoring.

Reply:
Got it, tx. Does this mean using `-version` or any illegal option, will now be simply ignored? Or would it allow outputting the version info on each build to the VS console?